src/components/ThreadList/ThreadList.js

```diff
@@ -20,72 +21,135 @@ class ThreadList extends Component {
 
         this.state = {
             userId: auth.getUserId(),
-            canDeleteThreads: auth.userHasPermission(permissions.deleteThread)
+            canDeleteThreads: auth.userHasPermission(permissions.deleteThread),
+            loading: false,
+            scrollPosition: 0
         }
     }
 
     componentDidMount() {
-        this.props.dispatch(threadActions.loadThreads());
+        this.props.dispatch(threadActions.enterThreadList(false));
+        window.addEventListener('scroll', debounce.debounce(this.handleScroll, 250));
```

Comment:
if we want to use the scroll watcher lets put a passive flag on it: https://developers.google.com/web/updates/2016/06/passive-event-listeners